[New] TimescaleDB on Kubernetes (#863)
timescale/docs: timescaledb/overview/core-concepts/timescale-kubernetes.md

# TimescaleDB on Kubernetes
You can install a TimescaleDB instance on any Kubernetes deployment. Use the `timescaledb-single` Helm chart to deploy a highly available TimescaleDB database, and the `timescaledb-multinode` Helm chart to deploy a multi-node distributed TimescaleDB database. 

> ## Highly available TimescaleDB 

Reviewer: ```suggestion
## High availability TimescaleDB 
```

The `timescaledb-single` Helm chart performs these actions:

> - Creates three default Pods using Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
> - Creates each Pod that has a container using the [TimescaleDB Docker image](https://github.com/timescale/timescaledb-docker-ha).
> - Installs TimescaleDB 2.1 and PG 13.
> - Creates containers that run a TimescaleDB instance and [Patroni](https://patroni.readthedocs.io/en/latest/) agent.
> - Configures each TimescaleDB instance for replication with one master and two replicas.

Reviewer: ```suggestion
* Creates three default Pods using Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
* Creates each Pod that has a container using the [TimescaleDB Docker image](https://github.com/timescale/timescaledb-docker-ha).
* Installs TimescaleDB 2.1 and PG 13.
* Creates containers that run a TimescaleDB instance and [Patroni](https://patroni.readthedocs.io/en/latest/) agent.
* Configures each TimescaleDB instance for replication with one master and two replicas.
```

<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-single.png" alt="Illustration of Highly Available TimescaleDB without backups to S3" width="640"/>

> When you deploy on AWS Elastic Kubernetes Service the Pods are scheduled on nodes which run in different Availability Zones (AZs). An AWS Elastic Load Balancer (ELB) is configured to handle routing incoming traffic to the master Pod.

Reviewer: ```suggestion
When you deploy on AWS Elastic Kubernetes service the Pods are scheduled on nodes which run in different availability zones (AZs). An AWS Elastic load balancer (ELB) is configured to handle routing incoming traffic to the master Pod.
```

<highlight type="note">
The backup to S3 is disabled by default. To configure backup location, credentials, schedules, and more, see the [Administrator Guide](https://github.com/timescale/timescaledb-kubernetes/blob/master/charts/timescaledb-single/admin-guide.md/).
</highlight>

When configured for backups to S3:
* Each Pod also includes a container running [pgBackRest](https://pgbackrest.org/).
* Two [cron jobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs) are created to handle full weekly and incremental daily backups.
* The backups are stored in an S3 bucket.

<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-single-backups.png" alt="Highly Available TimescaleDB with backups to S3" width="640"/>

## Multi-node distributed TimescaleDB
The `timescaledb-multinode` Helm chart deploys a multinode TimescaleDB cluster that sets up:

> - Creates s single TimescaleDB `Access Node` using Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
> - Creates three Pods containing `Data Nodes`using another Kubernetes StatefulSet.
> - Creates each Pod that has a container using a Docker image which includes the TimescaleDB multinode sources.

Reviewer: ```suggestion
* Creates s single TimescaleDB `Access Node` using Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
* Creates three Pods containing `Data Nodes`using another Kubernetes StatefulSet.
* Creates each Pod that has a container using a Docker image which includes the TimescaleDB multi-node sources.
```

<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-multi.png" alt="Multi-node TimescaleDB on Kubernetes" width="640"/>

> When you deploy on AWS Elastic Kubernetes Service, an AWS Elastic Load Balancer (ELB) is configured to handle routing incoming traffic to the access node.

Reviewer: ```suggestion
When you deploy on AWS Elastic Kubernetes service, an AWS Elastic load balancer (ELB) is configured to handle routing incoming traffic to the access node.
```

## Highly avaiable TimescaleDB 

The `timescaledb-single` Helm chart does the following:

- Three default Pods are created using a Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
- Each Pod has a container created using the [TimescaleDB Docker image](https://github.com/timescale/timescaledb-docker-ha).
- TimescaleDB 2.1 and PG 13 are installed.
- Each of the containers runs a TimescaleDB instance and [Patroni](https://patroni.readthedocs.io/en/latest/) agent.
- Each TimescaleDB instance is configured for replication  with one master and two replicas.

<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-single.png" alt="Illustration of Highly Available TimescaleDB without backups to S3" width="640"/>

When you deploy on AWS Elastic Kubernetes Service:
- The Pods are scheduled on nodes which run in different Availability Zones (AZs).

> - An AWS Elastic Load Balancer (ELB) is configured to handle routing incoming traffic to the master Pod.

Author: Done!!!

<highlight type="note">
The backup to S3 is disabled by default. To configure backup location, credentials, schedules, and more, see the [Administrator Guide](https://github.com/timescale/timescaledb-kubernetes/blob/master/charts/timescaledb-single/admin-guide.md/).
</highlight>

When configured for backups to S3:
- Each Pod also includes a container running [pgBackRest](https://pgbackrest.org/).
- Two [CronJobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs) are created to handle full weekly and incremental daily backups.
- The backups are stored to an S3 bucket.

> <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-single-backups.png" alt="Illustration of Highly Available TimescaleDB with backups to S3" width="640"/>

Reviewer: ```suggestion
<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-single-backups.png" alt="Highly Available TimescaleDB with backups to S3" width="640"/>
```

## Multi-node distributed TimescaleDB

> 

Reviewer: ```suggestion
```

> The `timescaledb-multinode` Helm chart deploys a multinode TimescaleDB cluster that does the following:

Reviewer: ```suggestion
The `timescaledb-multinode` Helm chart deploys a multinode TimescaleDB cluster that sets up:
```

- A single TimescaleDB `Access Node` using a Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) is created.
- Three Pods containing `Data Nodes`using another Kubernetes StatefulSet are created.

> - Each Pod has a container created using a Docker image which includes the TimescaleDB multinode sources.

Reviewer: ```suggestion
* A single TimescaleDB access node using a Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
* Three Pods containing data nodes using another Kubernetes StatefulSet.
* A container for aach Pod using a Docker image, which includes the TimescaleDB multi-node sources.
```

> <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-multi.png" alt="Illustration of Multi-node TimescaleDB on Kubernetes" width="640"/>

Reviewer: ```suggestion
<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-multi.png" alt="Multi-node TimescaleDB on Kubernetes" width="640"/>
```

> 

Reviewer: ```suggestion
```

> When you deploy on AWS Elastic Kubernetes Service:
> - An AWS Elastic Load Balancer (ELB) is configured to handle routing incoming traffic to the `Access Node`.

Reviewer: ```suggestion
When you deploy on AWS Elastic Kubernetes Service, an AWS Elastic Load Balancer (ELB) is configured to handle routing incoming traffic to the access node.
```

# TimescaleDB on Kubernetes

> You can install a TimescaleDB instance on any Kubernetes deployment. Use the `timescaledb-single` Helm chart to deploy a highly-available TimescaleDB database, and `timescaledb-multinode` Helm chart to deploy a multi-node distributed TimescaleDB database. 

Reviewer 2: ```suggestion
You can install a TimescaleDB instance on any Kubernetes deployment. Use the `timescaledb-single` Helm chart to deploy a highly available TimescaleDB database, and the `timescaledb-multinode` Helm chart to deploy a multi-node distributed TimescaleDB database. 
```

> ## A Highly Avaiable TimescaleDB 

Reviewer 2: ```suggestion
## Highly avaiable TimescaleDB 
```

The `timescaledb-single`Helm chart does the following:

- Creates three default pods using a Kubernetes [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
- Each pod has a container created using the [TimescaleDB Docker image](https://github.com/timescale/timescaledb-docker-ha).
  - TimescaleDB 2.1 and PG 13
- Each of the containers runs a TimescaleDB instance and [Patroni](https://patroni.readthedocs.io/en/latest/) agent.
- Each TimescaleDB instance is configured for replication  with one master and two replicas.

<img src="./timescaledb-single.png" width="640" />

When you deploy on AWS EKS:
- The pods are scheduled on nodes which run in different Availability Zones (AZs).
- An AWS Elastic Load Balancer (ELB) is configured to handle routing incoming traffic to the Master pod.

<highlight type="note">

> The backup to S3 is disabled by default, look at the

Reviewer 2: ```suggestion
The backup to S3 is disabled by default. See the
```

[Administrator Guide](admin-guide.md#backups) on how to configure backup location, credentials, schedules, and others.
</highlight>

When configured for backups to S3:
- Each pod also includes a container running [pgBackRest](https://pgbackrest.org/).
- By default, two [CronJobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) are created to handle full weekly and incremental daily backups.
- The backups are stored to an S3 bucket.

<img src="./timescaledb-single-backups.png" width="640" />

> ## A Multi Node Distributed TimescaleDB

Reviewer 2: ```suggestion
## Multi-node distributed TimescaleDB
```

> # TimescaleDB on Kubernetes

Reviewer 3: Sorry, can't make a file comment that I can find.

Without being able to easily view a preview of this PR because it's coming from a Forked repo, I can't confirm - but I think this file is in the wrong place. If you want to make it a child of "Deployment Options", then a new folder has to be created and the file placed there. As it is, we would expect to find this listed under "Core Concepts" which it's not.

  image](https://github.com/timescale/timescaledb-docker-ha).
* Installs TimescaleDB 2.1 and PG 13.
* Creates containers that run a TimescaleDB instance and
  [Patroni](https://patroni.readthedocs.io/en/latest/) agent.
* Configures each TimescaleDB instance for replication with one master and two
  replicas.

<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/timescaledb-single.png" alt="Illustration of Highly Available TimescaleDB without backups to S3" width="640"/>

When you deploy on AWS Elastic Kubernetes service the Pods are scheduled on
nodes which run in different availability zones (AZs). An AWS Elastic load
balancer (ELB) is configured to handle routing incoming traffic to the master
Pod.

<highlight type="note">
The backup to S3 is disabled by default. To configure backup location,
credentials, schedules, and more, see the [Administrator
Guide](https://github.com/timescale/timescaledb-kubernetes/blob/master/charts/timescaledb-single/admin-guide.md/).
</highlight>

When configured for backups to S3:
* Each Pod also includes a container running
  [pgBackRest](https://pgbackrest.org/).

> * Two [cron
>   jobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs) are
>   created to handle full weekly and incremental daily backups.

Reviewer: ```suggestion
* Two [cron jobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs) 
are created to handle full weekly and incremental daily backups.
```
